packages/flutter/lib/src/painting/strut_style.dart

```diff
@@ -0,0 +1,500 @@
+/// tall as the half leading plus descent.
+/// 
+/// ### Fields and their default values.
+
+// ///////////////////////////////////////////////////////////////////////////
+// The defaults are noted here for convenience. The actual place where they //
+// are defined is in the engine paragraph_style.h of LibTxt. The values here//
+// should be updated should it change in the engine. The engine specifies   //
+// the defaults in order to reduce the amount of data we pass to native as  //
+// strut will usually be unspecified.                                       //
+// ///////////////////////////////////////////////////////////////////////////
+
+///
+/// Omitted or null properties will take the default values specified below:
+///
+///  * [fontFamily] the name of the font to use when calcualting the strut (e.g., Roboto).
+///    No glyphs from the font will be drawn and the font will be used purely for metrics.
+///
+///  * [fontFamilyFallback] an ordered list of font family names that will be searched for when
+///    the font in [fontFamily] cannot be found. When all specified font families have been
+///    exhausted an no match was found, the default platform font will be used.
+///
+///  * [fontSize] the size of the ascent plus descent in logical pixels. This is also
+///    used as the basis of the custom leading caluclation. This value cannot
+///    be negative.
+///    Default is `14` logical pixels.
+///
+///  * [height] the multiple of [fontSize] to multiply the ascent and descent by.
+///    The [height] will impact the spacing above and below the baseline differently
+///    depending on the ratios between the font's ascent and descent. This property is
+///    separate from the leading multiplier, which is controlled through [leading].
+///    Default is `1.0`.
+///
+///  * [leading] the custom leading to apply to the strut as a multiple of [fontSize].
+///    Leading is additional spacing between lines. Half of the leading is added
+///    to the top and the other half to the bottom of the line height. This differs
+///    from [height] since the spacing is equally distributed above and below the
+///    baseline.
+///    Default is `null`, which will use the font-specified leading.
+///
+///  * [fontWeight] the typeface thickness to use when calculating the strut (e.g., bold).
+///    Default is `w400`.
+///
+///  * [fontStyle] the typeface variant to use when calculating the strut (e.g., italic).
+///    Default is `normal`.
+///
+///  * [forceStrutHeight] when true, all lines will be laid out with the height of the
+///    strut. All line and run-specific metrics will be ignored/overridden and only strut
+///    metrics will be used instead. This property guarantees uniform line spacing, however
+///    text in adjacent lines may overlap. This property should be enabled with caution as
+///    it bypasses a large portion of the vertical layout system.
+///    The default value is false.
```

Comment:
Is this equivalent to setting `height` and `leading` to zero for all TextSpans?

Reply:
Yes, although this flag is more performant than setting height to zero as it skips a lot of the per-run height logic. Another benefit is that it provides a single location where this can be controlled. It can be confusing and un-intuitive for users to be making changes to many TextStyles to adjust something that is controlled by strut. I would expect that the "set to zero" equivalency is not immediately obvious to most people learning this API.

There is also not yet a TextStyle leading property. The leading is currently bundled into the TextStyle.height property.